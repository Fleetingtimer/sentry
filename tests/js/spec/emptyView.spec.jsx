import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import EmptyView from 'app/components/emptyView';

describe('EmptyView', function() {
  it('renders', function() {
    let wrapper = shallow(<EmptyView>Empty View</EmptyView>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders w/ multiple children', function() {
    let wrapper = shallow(
      <EmptyView>
        Empty View
        <span>
          Another Child
        </span>
      </EmptyView>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with different icon', function() {
    let wrapper = shallow(
      <EmptyView icon={<span className="icon icon-test" />}>
        Empty View w/ icon
        <span>
          Another Child
        </span>
      </EmptyView>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
