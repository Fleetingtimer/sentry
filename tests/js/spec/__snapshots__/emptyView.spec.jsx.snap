// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyView renders 1`] = `
<div
  className="box empty-stream"
>
  <span
    className="icon icon-exclamation"
  />
  <p>
    Empty View
  </p>
</div>
`;

exports[`EmptyView renders w/ multiple children 1`] = `
<div
  className="box empty-stream"
>
  <span
    className="icon icon-exclamation"
  />
  <p>
    Empty View
  </p>
  <p>
    <span>
      Another Child
    </span>
  </p>
</div>
`;

exports[`EmptyView renders with different icon 1`] = `
<div
  className="box empty-stream"
>
  <span
    className="icon icon-test"
  />
  <p>
    Empty View w/ icon
  </p>
  <p>
    <span>
      Another Child
    </span>
  </p>
</div>
`;
